share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorShard.java

```diff
@@ -219,6 +232,8 @@ private void handleShareSnapshot(ShareSnapshotKey key, ShareSnapshotValue value)
                 snapshotUpdateCount.put(mapKey, 0);
             }
         }
+
+        offsetsManager.updateState(mapKey, offset);
```

Comment:
My understanding is that you don't use a timeline data structure in the offset manager. The risk is that a snapshot maybe replayed and hence updates the offset here. However, at this point, we don't know whether the snapshot will be durably stored. Let's imagine the worst case, the write of the snapshot fails for whatever reason. In this case, the snapshot registry is reverted to undo all the uncommitted changes related to the write. Hence all the state will be reverted but the state in the offset manager. Now, another operation may come before the snapshot is retried so your offset would be completely wrong now. It seems to me that there is a risk of trimming the log at a wrong offset due to this.

Reply:
will revert to timeline hm